tripal/tripal
# Add unpublish orphaned entities functionality

Issue #649 

## Type(s) of Change(s)


## Description


This is an implementation of the unpublish orphaned entities functionality.

## Testing?





1. Pull the branch and clear menu cache
1. Visit Content -> Tripal Content -> Delete Orphaned Entities (should be next to the publish link)
1. Select a bundle to inspect for orphaned entities.
1. If the bundle has orphaned entities, It'll provide you with the number of entities that should be removed and an example consisting of 10 entities that will get deleted.

If you need to manually create an orphaned entity you can:
1. create an organism entity using the GUI
1. delete the organism from chado using SQL.

## Screenshots (if appropriate):
![screen shot 2018-09-24 at 1 03 36 pm](https://user-images.githubusercontent.com/1512664/45966846-4844d580-bffa-11e8-914d-9914014b8786.png)

## Additional Notes (if any):



Note that the queries used are optimized to handle millions of records. This was tested on our live site and managed to analyze 5 million+ entities in under 2 seconds. However, the deleting job will be slow since we decided that it's best to trigger all `entity_delete` hooks which required that we process each entity individually.

Thanks!
